Do not try to write to ROS_ROOT for .rosstack_cache: this would leave dandling
files after package removal. Since the strategy is already to look into
${HOME}/.ros, better use this systematically.

--- tools/rospack/rosstack.cpp~	2011-05-25 02:14:00.000000000 +0200
+++ tools/rospack/rosstack.cpp	2012-01-18 13:43:22.058510238 +0100
@@ -824,11 +824,13 @@
 string ROSStack::getCachePath()
 {
   string path;
+#if 0
   path = string(ros_root) + fs_delim + ".rosstack_cache";
   if (access(ros_root, W_OK) == 0)
     return path;
   // if we cannot write into the ros_root, then let's try to
   // write into the user's .ros directory.
+#endif
   createROSHomeDirectory();
   path = string(getenv("HOME")) + fs_delim + ".ros" + fs_delim + "rosstack_cache";
   return path;
